rust/chains/hyperlane-ethereum/src/contracts/multicall.rs

```diff
@@ -0,0 +1,52 @@
+use std::sync::Arc;
+
+use ethers::{abi::Detokenize, providers::Middleware};
+use ethers_contract::{builders::ContractCall, Multicall, MulticallResult, MulticallVersion};
+use hyperlane_core::{
+    utils::hex_or_base58_to_h256, ChainResult, HyperlaneDomain, HyperlaneProvider,
+};
+
+use crate::{ConnectionConf, EthereumProvider};
+
+const ALLOW_BATCH_FAILURES: bool = true;
+
+pub async fn build_multicall<M: Middleware + 'static>(
+    provider: Arc<M>,
+    conn: &ConnectionConf,
+    domain: HyperlaneDomain,
+) -> eyre::Result<Multicall<M>> {
+    let address = conn
+        .message_batch
+        .multicall3_address
+        .unwrap_or(hex_or_base58_to_h256("0xcA11bde05977b3631167028862bE2a173976CA11").unwrap());
```

Comment:
nit: don't feel strongly, but i kinda feel this was more appropriate in the settings as it was before - that way it's easier at a glance to see what the default value is as it's defined where the setting is defined